bindings/python/src/MapObjective.cpp

```diff
@@ -0,0 +1,52 @@
+#include "CommonPybindUtilities.h"
+#include "MParT/Distributions/GaussianSamplerDensity.h"
+#include "MParT/MapObjective.h"
+#include <pybind11/stl.h>
+#include <pybind11/eigen.h>
+
+#include <Kokkos_Core.hpp>
+#include <pybind11/pybind11.h>
+
+namespace py = pybind11;
+using namespace mpart;
+using namespace mpart::binding;
+
+template<typename MemorySpace>
+void mpart::binding::MapObjectiveWrapper(py::module &m) {
+    std::string moName= "MapObjective";
+    std::string tName = "GaussianKLObjective";
+    if(!std::is_same<MemorySpace,Kokkos::HostSpace>::value) {
+        moName = "d" + tName;
+        tName = "d" + tName;
+    }
+
+    py::class_<MapObjective<MemorySpace>, std::shared_ptr<MapObjective<MemorySpace>>>(m, moName.c_str())
+        .def("TestError", &KLObjective<MemorySpace>::TestError)
+        .def("TrainError", &KLObjective<MemorySpace>::TrainError)
+    ;
+
+    py::class_<KLObjective<MemorySpace>, MapObjective<MemorySpace>, std::shared_ptr<KLObjective<MemorySpace>>>(m, tName.c_str())
+        .def(py::init( [](Eigen::Ref<Eigen::MatrixXd> &train){
+            StridedMatrix<double, MemorySpace> trainView = MatToKokkos<double, MemorySpace>(train);
+            Kokkos::View<double**,MemorySpace> storeTrain ("Training data store", trainView.extent(0), trainView.extent(1));
+            Kokkos::deep_copy(storeTrain, trainView);
+            std::shared_ptr<GaussianSamplerDensity<MemorySpace>> density = std::make_shared<GaussianSamplerDensity<MemorySpace>>(trainView.extent(0));
```

Comment:
Like for the matlab bindings, I don't think we should restrict ourselves to a standard normal reference distribution in the bindings if the c++ code is more general.